### PR TITLE
Feat: Add Application Update

### PR DESCRIPTION
**Pull request by Mubangizi**
#### What this PR does
- Adds app update routes
- Adds `replicas` and `revision` metadata fields to the app detail view
- Add deletion of `secrets` if the application is deleted

#### How it can be tested
- They can be tested with `PATCH` on route `/projects/{project_id}/apps/{app_id}` for users and `/apps/{app_id}` for admins

### CODE REVIEW

**Review by ttcollins — APPROVED** [review]
Great work @Mubangizi

File: app/controllers/app.py [hunk 1]
```diff
@@ -934,6 +940,378 @@ def delete(self, app_id):
         except Exception as e:
             return dict(status='fail', message=str(e)), 500
 
+    
+    @admin_required
```
Comment (inno-asiimwe) [inline app/controllers/app.py:944]: This does not need to been an admin only resource.

Comment (MutegekiHenry) [reply app/controllers/app.py:944]: Seems like Allan was trying to separate the admin and user endpoints

**Review by MutegekiHenry — APPROVED** [review]
Great work @Mubangizi

**Review by inno-asiimwe — CHANGES_REQUESTED** [review]
Let us do the app update on how the original apps were setup, Incase we need to change the design and flow of routes, let us put that in a different ticket and discuss it before we can implement it.

File: app/controllers/app.py [hunk 2]
```diff
@@ -934,6 +940,378 @@ def delete(self, app_id):
...
+                message=f'App updated successfully'
+                ), 200
+
+        except client.rest.ApiException as exc:
+            return dict(status='fail', message=exc.reason), exc.status
+
+        except Exception as exc:
+            return dict(status='fail', message=str(exc)), 500
+
+class ProjectAppsDetailView(Resource):
```
[the remark this answers is not in the document]
Comment (Mubangizi) [reply app/controllers/app.py]: Hey @inno-asiimwe , this has been resolved

**Review by inno-asiimwe — APPROVED** [review]

**Pull request merged**